network/dag/state.go

```diff
@@ -284,32 +312,38 @@ func (s *state) Start() error {
 		return err
 	}
 
-	if err := s.Verify(ctx); err != nil {
+	if err := s.Verify(); err != nil {
 		return err
 	}
 
 	return nil
 }
 
 func (s *state) Statistics(ctx context.Context) Statistics {
-	return s.graph.Statistics(ctx)
+	return s.graph.statistics(ctx)
 }
 
-func (s *state) Verify(ctx context.Context) error {
-	transactions, err := s.FindBetween(ctx, MinTime(), MaxTime())
-	if err != nil {
-		return err
-	}
-	for _, tx := range transactions {
-		if err := s.verifyTX(ctx, tx); err != nil {
+func (s *state) Verify() error {
+	return storage.BBoltTXView(context.Background(), s.db, func(contextWithTX context.Context, tx *bbolt.Tx) error {
+		transactions, err := s.graph.findBetween(tx, MinTime(), MaxTime())
+		if err != nil {
 			return err
 		}
-	}
-	return nil
+		for _, tx := range transactions {
+			if err := s.verifyTX(contextWithTX, tx); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
 }
 
 func (s *state) Walk(ctx context.Context, visitor Visitor, startAt hash.SHA256Hash) error {
-	return s.graph.Walk(ctx, visitor, startAt)
+	return storage.BBoltTXUpdate(ctx, s.db, func(contextWithTX context.Context, tx *bbolt.Tx) error {
```

Comment:
This should be a read TX? Otherwise we always lock the db.

Reply:
Is called by migrate which does an update. I'll change migrate so it starts a transaction and calls the private walk.

Reply:
actually not needed, so changed to read tx